### PR TITLE
CPP2-20 Communicate with RGB sensor

### PR DESCRIPTION
**Pull request by kapeterakos**

### CODE REVIEW

File: NUCLEO-144_app/.cproject [hunk 1]
```diff
@@ -84,7 +84,7 @@
 								<option id="fr.ac6.managedbuild.tool.gnu.cross.c.linker.script.1397397055" name="Linker Script (-T)" superClass="fr.ac6.managedbuild.tool.gnu.cross.c.linker.script" useByScannerDiscovery="false" value="../STM32F746ZGTx_FLASH.ld" valueType="string"/>
 								<option id="gnu.c.link.option.libs.636332255" name="Libraries (-l)" superClass="gnu.c.link.option.libs" useByScannerDiscovery="false"/>
 								<option id="gnu.c.link.option.paths.933961" name="Library search path (-L)" superClass="gnu.c.link.option.paths" useByScannerDiscovery="false"/>
-								<option id="gnu.c.link.option.ldflags.264614128" name="Linker flags" superClass="gnu.c.link.option.ldflags" useByScannerDiscovery="false" value="-specs=nosys.specs -specs=nano.specs" valueType="string"/>
+								<option id="gnu.c.link.option.ldflags.264614128" name="Linker flags" superClass="gnu.c.link.option.ldflags" useByScannerDiscovery="false" value="-u _printf_float" valueType="string"/>
```
Comment (LeinadSokram) [inline NUCLEO-144_app/.cproject]: It seems to me that there's some formatting error :)

Comment (johnyka) [inline NUCLEO-144_app/.cproject]: please don't use it. It might cause random crash

File: NUCLEO-144_app/.cproject [hunk 2]
```diff
@@ -252,9 +252,10 @@
 	</storageModule>
 	<storageModule moduleId="org.eclipse.cdt.core.LanguageSettingsProviders"/>
 	<storageModule moduleId="refreshScope" versionNumber="2">
-		<configuration artifactName="${ProjName}" configurationName="Debug">
```
Comment (johnyka) [inline NUCLEO-144_app/.cproject]: I don't really understand this change. Could you revert please?

File: NUCLEO-144_app/Src/main.c [hunk 3]
```diff
@@ -1455,13 +1561,14 @@ void StartDefaultTask(void const * argument)
     /* Turning on BT Communication */
     HAL_GPIO_WritePin(GPIOG, BT_nRESET_Pin, GPIO_PIN_SET);
 
-    printf("Welcome to the CPP2 project!\r\n");
-    float i = 935;
-    htim1.Instance->CCR1 = i;
+
+        float i = 935;
```
Comment (johnyka) [inline NUCLEO-144_app/Src/main.c]: Too many spaces here

File: NUCLEO-144_app/Src/main.c [hunk 4]
```diff
@@ -1455,13 +1561,14 @@ void StartDefaultTask(void const * argument)
...
+        htim1.Instance->CCR1 = i;
 
     /* Infinite loop */
     for (;;) {
-
+        printf("Welcome to the CPP2 project!\r\n");
```
Comment (johnyka) [inline NUCLEO-144_app/Src/main.c]: Why did you move this here?

File: NUCLEO-144_app/Src/main.c [hunk 5]
```diff
@@ -83,6 +83,43 @@ typedef enum motor_direction {
...
+
+
+
+uint8_t temp_reg_address = 0;
+uint8_t temp_data_to_read;
+
+char color_detected, color_flag;
+
+// Find maximal floating point value
+#define MAX_FLOAT(a, b) (((a) > (b)) ? (a) : (b))
```
Comment (johnyka) [inline NUCLEO-144_app/Src/main.c]: Where do you use this?
